File: python/selectionSort.py
def selectionSort(arr):
    n = len(arr)
    for i in range(n):
        min_index = i
        for j in range(i + 1, n):
            if arr[j] < arr[min_index]:
                min_index = j

        arr[i], arr[min_index] = arr[min_index], arr[i]

    return arr


arr = []
x = int(input("Enter the no of elements of the array : "))
print("Enter the elements of array one by one : ")

for i in range(x):
    n = int(input())
    arr.append(n)

sorted_arr = selectionSort(arr)
print("Sorted array is:", sorted_arr)

File: data structure in c/selectionSort.c
#include<stdio.h>

void selectionSort(int arr[] , int x){
    int minIndex , temp;
    for (int i = 0; i < x; i++){
        minIndex = i;
        for (int j = i + 1; j < x; j++){
            if (arr[j] < arr[minIndex]){
                minIndex = j;
            }
            
        }
        temp = arr[minIndex];
        arr[minIndex] = arr[i];
        arr[i] = temp;
        
    }
    
}

void main(){
    int size;
    printf("Enetr the size of array : ");
    scanf("%d", &size);
    int arr[size];
    printf("\nenter array elements...\n");
    for (int i = 0; i < size; i++){
        scanf("%d", &arr[i]);
    }

    selectionSort(arr,size);

    printf("printing array elements....\n");
    for (int i = 0; i < size; i++){
        printf("%d  ", arr[i]);
    }
    
    
}